app/jobs/notify_webhook_job.rb

```diff
@@ -29,10 +33,4 @@ def perform(notification_id:)
       raise e # re-raise the error to force the notification to be retried by sidekiq later
     end
   end
-
-private
-
-  def client
```

Comment:
I think I prefer this as a separate method TBH - makes the main part of the code easier to read.

Reply:
You could perhaps pass `subscription` as an argument to this method and move the basic auth header assignment stuff (lines 18-20) to here:

```ruby
def client(subscription)
  Faraday.new(headers: { 'Content-Type': 'application/vnd.api+json', 'User-Agent': 'pecs-webhooks/v1' }).tap do |client|
    if subscription.username.present? && subscription.password.present?
      client.headers['Authorization'] = "Basic #{Base64.strict_encode64(subscription.username + ':' + subscription.password)}"
    end
  end
end
```

Reply:
Ok - i've done this minus the `tap`